joblib/test/test_parallel.py

```diff
@@ -289,6 +288,13 @@ def test_exception_dispatch():
                     (delayed(exception_raiser)(i) for i in range(30)),
             )
 
+def test_nested_exception_dispatch():
+    """Ensure TransportableException objects for nested joblib cases gets propagated.
```

Comment:
Personally I generally resist the temptation of adding docstrings to test functions and use comments instead. 

The thing I find slightly annoying is that the docstring gets printed when nosetests in versbose mode (verbose is enabled by default for joblib in setup.cfg). It creates a minor friction that you then have to figure out the failing test function name from the docstring, for example when you want to run only the failing function (for example by using `-m test_the_function_that_is_failing --ipdb-f` to drop into a ipdb session and understand the failure better).

More than happy to hear good arguments for using docstrings. cc @ogrisel @GaelVaroquaux in case they have strong opinions on this one.